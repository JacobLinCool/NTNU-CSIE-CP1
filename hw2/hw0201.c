#include <stdio.h>
#include <stdint.h>

int main() {
    // ax^2 + bx + c = 0
    int32_t a, b, c;
    printf("Please enter a quadratic polynomial (a,b,c): ");
    // Throw an ERROR if the input is not accepted
    if (scanf("%d,%d,%d", &a, &b, &c) != 3 || a == 0) {
        printf("Invalid Input!\n");
        return 1;
    }

    // Calculate discriminant
    int32_t d = b * b - 4 * a * c;

    if (d < 0) printf("No real roots.\n");
    else if (d == 0) printf("One real root.\n");
    else printf("Two distinct real roots.\n");

    return 0;
}
