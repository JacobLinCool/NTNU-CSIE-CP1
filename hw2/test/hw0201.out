Please enter a quadratic polynomial (a,b,c): Two distinct real roots.
[[===]]
Please enter a quadratic polynomial (a,b,c): One real root.
[[===]]
Please enter a quadratic polynomial (a,b,c): No real roots.
